bin/src/hickory-dns.rs

```diff
@@ -123,6 +131,16 @@ struct Cli {
     #[clap(long = "quic-port", value_name = "QUIC-PORT")]
     pub(crate) quic_port: Option<u16>,
 
+    /// Listening socket for Prometheus metrics,
+    /// defaults to 127.0.0.1:9000, for remote access configure socket as needed (e.g. 0.0.0.0:9000)
+    /// overrides any value in config file
+    #[cfg(feature = "__prometheus")]
+    #[clap(
+        long = "prometheus-listen-address",
+        value_name = "PROMETHEUS-LISTEN-ADDRESS"
+    )]
+    pub(crate) prometheus_listen_addr: Option<SocketAddr>,
```

Comment:
I think we can set a default with a clap attribute so that we can avoid the `Option` wrapper here?

Reply:
Updated.

Reply:
Reverted, makes it impossible to decide if config file was overrriden by cli args or is default value and if parameter from config file should be used.